resources/design-document.md

```diff
@@ -0,0 +1,244 @@
+
+U4. As a BTM customer, I want to be able to create tests for trainings
+
+U5. As a BTM customer, I want to be able to view all active trainings that are close to expiration
+
+U6. As a BTM customer, I want to be able to update test results for retests
+
+U7. As a BTM customer, I want to be able to view all trainings done within a series
+
+U8. As a BTM customer, I want to be able to view specific trainings
+
+U9. As a BTM customer, I want to be able to view all employees on a specific team
+
+U10. As a BTM customer, I want to be able to view all tests taken by an employee
+
+U11. As a BTM customer, I want to be able to deactivate employees that have left
+
+
+Stretch goals
+
+S1. As a BTM customer, I want to have my trainings auto-update their status based off of their date and months to expiration
+
+S2. As a BTM customer, I want to be able to make the next training in a training series by copying the information from the most recent training
+
+
+## 3. Project Scope
+
+### 3.1. In Scope
+
+* Creating, retrieving, and updating an employee
+* Test list that can be searched by training or employee name
+* Recipes can be retrieved and individual recipes added to meal plan
+* A user's meal plan which consists of chosen recipes
+
+### 3.2. Out of Scope
+
+* Trainings auto-updating their status
+* Training Series duplicator for next in series
+
+# 4. Proposed Architecture Overview
+
+This initial iteration will provide the minimum lovable product (MLP) including
+creating, retrieving, and updating an employee, as well as retrieving a training and creating testing for it; and creating a Training Series.
+
+We will use API Gateway and Lambda to create 15(20) endpoints (defined in section 6.2 "Endpoints")
+that will handle the creation, update, and retrieval of Employees, Trainings, Tests and Training Series to satisfy our
+requirements.
+
+We will store employees, trainings, tests, and training series in their own tables in DynamoDB.
+
+Brecghan's Training Matrix(BTM) will also provide a web interface for users to manage
+the matrix. A main page providing connections to view trainings, employees, or test results. From those pages Users
+will be able to create new items and link off to the other viewing pages.
+
+# 5. API
+
+## 6.1. Public Models
+
+```
+// EmployeeModel
+
+String employeeId;
+String employeeName;
+Boolean isActive;
+Enum team;
+Date startDate;
+Set<String> trainingsTaken;
+Set<String> testsTaken;
+Enum trainingStatus;
+```
+
+```
+// TrainingModel
+
+String trainingId;
+String trainingName;
+ZoneDateTime trainingDate;
+Boolean isActive;
+Integer monthsTilExpire;
+Enum expirationStatus;
+Set<String> testsForTraining;
+Set<String> employeesTrained;
+String trainingSeries;
+
+```
+
+```
+// TestModel
+
+String trainingId;
+String employeeId;
+Integer scoreToPass;
+Integer latestScore;
+Boolean hasPassed;
+Integer timesTaken;
+```
+
+```
+// TrainingSeriesModel
+
+String trainingSeriesName;
+```
+
+## 6.2. Endpoints
```

Comment:
A formatting note, but it's actually helpful ... can you add a heading for each one of the endpoints with the name of it? E.g. like is done in the [example](https://github.com/nss-se-cohort-3/u5-ProjectTemplate/blob/main/resources/example-design-document.md?plain=1#L121)